src/libraries/Microsoft.Extensions.Logging/tests/Common/LoggerMessageTest.cs

```diff
@@ -322,6 +323,33 @@ public void DefineScope_WithNoParameters_ThrowsException_WhenFormatString_HasNam
             Assert.Equal(expectedMessage, exception.Message);
         }
 
+        [Theory]
+        [MemberData(nameof(DefineMethodsData))]
+        public void DefineMessage_ThrowsException_WhenFormatString_IsNull(Delegate define)
+        {
+            // Arrange
+            var expectedMessage = $"Value cannot be null. (Parameter 'format')";
+            var parameters = new object[] { LogLevel.Error, new EventId(0), null };
+
+            // Act
+            var exception = Assert.Throws<TargetInvocationException>(() => define.DynamicInvoke(parameters));
+
+            // Assert
+            Assert.IsType<ArgumentNullException>(exception.InnerException);
+            Assert.Equal(expectedMessage, exception.InnerException.Message);
```

Comment:
>expectedMessage [](start = 25, length = 15)

It is not a good practice to test the exception message. These message can get localized at some point and will cause such tests to fail. 

It is enough to check the exception.ParamName property.

Reply:
Thank you for taking the time to look. I doubted whether it was good to test the message, but I just didn't want to be inconsistent with other tests in the suit which do that. 

I was also wondering it it was good to check `ParamName` property? I mean we can consider the `format` parameter of the method as an implementation detail. Do we want to change test each time the param name has been changed? Or are you suggesting to get a param name of the method via reflection and test the exception.ParamName against that?